generated/nidaqmx/_grpc_interpreter.py

```diff
@@ -2233,9 +2233,10 @@ def read_analog_f64(
         response = self._invoke(
             self._client.ReadAnalogF64,
             grpc_types.ReadAnalogF64Request(
-                fill_mode_raw=fill_mode,
-                num_samps_per_chan=num_samps_per_chan, read_array=read_array,
-                task=task, timeout=timeout))
+                array_size_in_samps=read_array.size, fill_mode_raw=fill_mode,
+                num_samps_per_chan=num_samps_per_chan, task=task,
+                timeout=timeout))
+        read_array[:]=response.read_array
         return response.read_array, response.samps_per_chan_read
```

Comment:
These are returning the wrong read array. They should return the numpy read array, not the gRPC read array.

Reply:
By "return the numpy read array", I meant do this:

```
read_array[:]=response.read_array
return read_array, response.samps_per_chan_read
```

You did that, but you also changed the code to allocate a temporary numpy array:
```
read_array[:] = numpy.array(response.read_array)
return read_array, response.samps_per_chan_read
```

The reason why some of the read functions accept a numpy array from the user is to avoid unnecessarily allocating and copying arrays. DAQmx applications may work with large amounts of data, so when working on DAQmx read/write functions, we should assume that allocations and copies are expensive.

The right hand side of `read_array[:] = ...` does not need to be a numpy array. It can be a list:
```
>>> import numpy
>>> read_array = numpy.zeros(3, dtype=numpy.double)
>>> read_array
array([0., 0., 0.])
>>> read_array[:] = [1.1,2.2,3.3]
>>> read_array
array([1.1, 2.2, 3.3])
```

Or a tuple:
```
>>> read_array[:] = (3.3,4.4,5.5)
>>> read_array
array([3.3, 4.4, 5.5])
```

Or a Protobuf repeated field:
```
>>> import nidaqmx._stubs.nidaqmx_pb2
>>> response = nidaqmx._stubs.nidaqmx_pb2.ReadAnalogF64Response()
>>> response.read_array.extend([6.6,7.7,8.8])
>>> response
read_array: 6.6
read_array: 7.7
read_array: 8.8

>>> read_array[:] = response.read_array
>>> read_array
array([6.6, 7.7, 8.8])
```